src/scenechange/mod.rs

```diff
@@ -240,7 +245,7 @@ impl<T: Pixel> SceneChangeDetector<T> {
     let mut result = if self.speed_mode == SceneDetectionSpeed::Fast {
       self.fast_scenecut(frame1, frame2)
```

Comment:
Ok, when fast_scenecut is computed you need to populate lookahead_intra_costs from scratch.